### PR TITLE
KHP3-5239: (refactor) migrate procedures order panel and order form to radiology app

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR migrates procedure related features to esm-procedure. Initial work done by @PatrickWaweru 

## Screenshots
https://github.com/palladiumkenya/esm-procedure-app/assets/28008754/244a67f3-54ab-4558-becb-5be16d8e5afb

### CODE REVIEW

File: src/config-schema.ts [hunk 1]
```diff
@@ -28,14 +11,34 @@ export const configSchema = {
...
+    _description: "List of lab orderable concepts",
+    _default: {
+      labOrderableConcepts: [],
+      labOrderTypeUuid: "",
+    },
+  },
 };
 
 export type Config = {
   casualGreeting: boolean;
```
Comment (ojwanganto) [inline src/config-schema.ts:25]: Where have we used these?

File: src/form/procedures-orders/add-procedures-order/add-procedures-order.workspace.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  type DefaultWorkspaceProps,
+  launchPatientWorkspace,
+  type OrderBasketItem,
+} from "@openmrs/esm-patient-common-lib";
+import { TestTypeSearch } from "./procedures-type-search";
+import { ProceduresOrderForm } from "./procedures-order-form.component";
+import styles from "./add-procedures-order.scss";
+import { type ProceduresOrderBasketItem } from "../../../types";
+
+export interface AddProceduresOrderWorkspaceAdditionalProps {
```
Comment (ojwanganto) [inline src/form/procedures-orders/add-procedures-order/add-procedures-order.workspace.tsx:24]: Ideally, this should read something like **AddProcedureOrderWorkspaceAdditionalProps**

File: src/form/procedures-orders/add-procedures-order/procedures-order.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+// See the Urgency enum in https://github.com/openmrs/openmrs-core/blob/492dcd35b85d48730bd19da48f6db146cc882c22/api/src/main/java/org/openmrs/Order.java
+export const priorityOptions = [
+  { value: "ROUTINE", label: "Routine" },
+  { value: "STAT", label: "Stat" },
+  { value: "SCHEDULED", label: "Scheduled" },
+];
+// TODO add priority option `{ value: "ON_SCHEDULED_DATE", label: "On scheduled date" }` once the form supports a date.
+
+export function createEmptyLabOrder(
```
Comment (ojwanganto) [inline src/form/procedures-orders/add-procedures-order/procedures-order.ts:12]: you may have been thinking about empty procedure order

File: src/form/procedures-orders/add-procedures-order/procedures-type-search.tsx [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+  useOrderBasket,
+} from "@openmrs/esm-patient-common-lib";
+import { prepProceduresOrderPostData } from "../api";
+import { type ProceduresType, useProceduresTypes } from "./useProceduresTypes";
+import { createEmptyLabOrder } from "./procedures-order";
+import styles from "./procedures-type-search.scss";
+import { type ProceduresOrderBasketItem } from "../../../types";
+
+export interface TestTypeSearchProps {
+  openLabForm: (searchResult: ProceduresOrderBasketItem) => void;
```
Comment (ojwanganto) [inline src/form/procedures-orders/add-procedures-order/procedures-type-search.tsx:34]: I would go for ProcedureOrderBasketItem

**Review by ojwanganto — CHANGES_REQUESTED** [review]
Thanks, @donaldkibet. I think we can generally favour ProcedureOrder instead of ProceduresOrder

File: src/form/procedures-orders/add-procedures-order/procedures-order.ts (hunk 3 above)
Comment (donaldkibet) [reply src/form/procedures-orders/add-procedures-order/procedures-order.ts:12]: Haven't thought about that, just moving the files over to esm-procedure-app

Comment (ojwanganto) [reply src/form/procedures-orders/add-procedures-order/procedures-order.ts:12]: I basically was referring to the name: **createEmptyLabOrder**. It should be something emptyProcedureOrder or something, ama?

**Comment by donaldkibet** [comment]
Merging this now since most of the requested changes have been addressed. Will address subsequent changes in another PR if need be. Thanks @ojwanganto

**Comment by ojwanganto** [comment]
Thanks @donaldkibet

**Pull request merged**